### PR TITLE
 Implemented Random Forest Classifier With RandomUnderSampler

### LINKED ISSUE #70: Improve Random Forest model
### Prerequisites

- [X] I am an Open Source Contributor
- [X] I checked to make sure that this issue has not already been filed

### Description

I want to do OverSampling and UnderSampling on Random Forest classifier.Please assign me this issue.

### Screenshots

_No response_

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/Rakesh9100/ML-Project-Drug-Review-Dataset/blob/main/.github/CONTRIBUTING_GUIDELINES.md) of this project.

### PR DESCRIPTION
**Pull request by sreyaad**
# Fixes Issue🛠️

Closes #70 

# Description👨‍💻 

- Random Forest Classifier with Random Under Sampler
# Type of change📄

- [] Bug fix (non-breaking change which fixes an issue)
- [✅] New feature (non-breaking change which adds functionality)

# How this has been tested✅

# Checklist✅ 

- [✅] My code follows the style guidelines of this project
- [✅] I have performed a self-review of my own code
- [✅] I have commented my code, particularly in hard-to-understand areas
- [✅] I have added demonstration in the form of GIF/video file
- [✅] I am an Open Source Contributor

# Screenshots/GIF📷
image

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -430,3 +432,31 @@ def decode_html(text):
...
+
+
+##### Random Forest Algorithm with Random UnderSampler #####
+sampler = RandomUnderSampler()
+X_train1, y_train1 = sampler.fit_resample(X_train, Y_train)
+
+clf2= RandomForestClassifier().fit(X_train1, y_train1)
+
+#Make Predictions
+pred = clf2.predict(X_test)
```
Comment (Neilblaze) [inline main.py:444]: I'll recommend using `pred1` here and updating the changes accordingly.

File: main.py [hunk 2]
```diff
@@ -430,3 +432,31 @@ def decode_html(text):
...
+print(confusion_matrix(pred, Y_test))
+
+
+##### Random Forest With SMOTE
+
+sampler2 = SMOTE()
+X_train2, y_train2 = sampler2.fit_resample(X_train, Y_train)
+clf3= RandomForestClassifier().fit(X_train2, y_train2)
+
+pred = clf3.predict(X_test)
```
Comment (Neilblaze) [inline main.py:458]: Similarly, use `pred2` here & update the changes accordingly.

**Review by Neilblaze — CHANGES_REQUESTED** [review]
Make the changes, and rest LGTM! Also, please remove the linting errors :)

**Comment by Rakesh9100** [comment]
@sreyaad Pls make the requested changes

**Comment by Rakesh9100** [comment]
Closing for no response.

**Pull request closed without merging**